samples/tacos/src/main/kotlin/com/slack/circuit/tacos/OrderTacosCircuit.kt

```diff
@@ -0,0 +1,356 @@
+  currentOrder: OrderDetails,
+): OrderDetails {
+  val newFilling =
+    when (event) {
+      is OrderStep.UpdateOrder.SetFilling -> event.ingredient
+      else -> currentOrder.filling
+    }
+  val newToppings =
+    when (event) {
+      is OrderStep.UpdateOrder.SetToppings -> event.ingredients
+      else -> currentOrder.toppings
+    }
+
+  val newIngredientsCost = calculateIngredientsCost(newFilling, newToppings)
+
+  return OrderDetails(
+    baseCost = currentOrder.baseCost,
+    ingredientsCost = newIngredientsCost,
+    filling = newFilling,
+    toppings = newToppings.toImmutableSet()
+  )
+}
+
+private fun calculateIngredientsCost(filling: Ingredient?, toppings: Set<Ingredient>): BigDecimal {
+  var cost = filling?.charge ?: BigDecimal.ZERO
+  toppings.forEach { topping -> cost += topping.charge }
+  return cost
+}
+
+@Composable
+internal fun OrderTacosUi(state: OrderTacosScreen.State, modifier: Modifier = Modifier) {
+  val sink = state.eventSink
+  Scaffold(
+    modifier = modifier.padding(5.dp),
+    topBar = {
+      CenterAlignedTopAppBar(
+        title = { Text(stringResource(state.headerResId)) },
+        modifier = Modifier,
+        navigationIcon = {
+          NavigationButton(
+            direction = Direction.LEFT,
+            visible = state.isPreviousVisible,
+          ) {
+            sink(OrderTacosScreen.Event.Previous)
+          }
+        },
+        actions = {
+          NavigationButton(
+            direction = Direction.RIGHT,
+            enabled = state.isNextEnabled,
+            visible = state.isNextVisible,
+          ) {
+            sink(OrderTacosScreen.Event.Next)
+          }
+        }
+      )
+    },
+    bottomBar = {
+      OrderTotal(
+        orderCost = state.orderCost,
+        onConfirmationStep = state.stepState is ConfirmationOrderStep.OrderState,
+        isVisible = state.isFooterVisible,
+      ) {
+        sink(OrderTacosScreen.Event.ProcessOrder)
+      }
+    }
+  ) { padding ->
+    val stepModifier = Modifier.padding(padding)
+    when (state.stepState) {
+      is FillingsOrderStep.State -> FillingsUi(state.stepState, stepModifier)
+      is ToppingsOrderStep.State -> ToppingsUi(state.stepState, stepModifier)
+      is ConfirmationOrderStep.OrderState -> ConfirmationUi(state.stepState, stepModifier)
+      is SummaryOrderStep.SummaryState -> SummaryUi(state.stepState, stepModifier)
+    }
+  }
+}
+
+internal enum class Direction(val icon: ImageVector, @StringRes val descriptionResId: Int) {
+  LEFT(Icons.Filled.ArrowBack, R.string.top_bar_back),
+  RIGHT(Icons.Filled.ArrowForward, R.string.top_bar_forward)
+}
+
+@Composable
+private fun NavigationButton(
+  direction: Direction,
+  modifier: Modifier = Modifier,
+  enabled: Boolean = true,
+  visible: Boolean = true,
+  onClick: () -> Unit
+) {
+  if (!visible) return
+
+  val tintColour =
+    when (enabled) {
+      true -> MaterialTheme.colorScheme.onSurface
+      false -> MaterialTheme.colorScheme.outline
+    }
```

Comment:
I think this means something else is wrong as you shouldn't need to do this manually